Remove most outdated version notes

parlough opened the pull request:
With this change I kept the version notes about the old versions of the `dart` tools for now as their deprecation is pretty recent and I still see a fair bit of usage of the old commands in the wild. Do you think those should be removed now as well @kwalrath?

Fixes https://github.com/dart-lang/site-www/issues/2174

kwalrath commented:
> With this change I kept the version notes about the old versions of the `dart` tools for now as their deprecation is pretty recent and I still see a fair bit of usage of the old commands in the wild. Do you think those should be removed now as well @kwalrath?

No, I think you're right to keep the tool notes in, for now.

kwalrath commented on src/_guides/language/extension-methods.md:
```diff
@@ -8,11 +8,6 @@ You might use extension methods without even knowing it.
-  the pubspec must have a [minimum SDK constraint][] of at least 2.7.0.
-{{site.alert.end}}
-
 [minimum SDK constraint]: /tools/pub/pubspec#sdk-constraints
```
Is this link used anywhere else? If not:

```suggestion
```

kwalrath reviewed (commented):
Aside from a couple of small things, LGTM!

kwalrath commented on src/tools/index.md:
```diff
@@ -19,11 +19,10 @@ If you aren't sure which tools you need, **get the Flutter SDK.**
-{{site.alert.version-note}}
-  As of Flutter 1.21, the Flutter SDK includes the full Dart SDK.
+{{site.alert.note}}
+The Flutter SDK includes the full Dart SDK.
```
It's our standard to indent the note text 2 spaces. Idk why; I just obey.

```suggestion
  The Flutter SDK includes the full Dart SDK.
```

parlough replied:
Oops, thanks! Lost them in the copy paste :)

parlough commented:
@kwalrath Thanks for the review. I've addressed your comments.

I've also created https://github.com/dart-lang/site-www/issues/3487 to track the future work on removing the old tool version notes.

kwalrath reviewed (approved):
lgtm. Thanks, @parlough!
